pkg/util/exithandler.go

```diff
@@ -28,22 +28,26 @@ func RegisterExitHandler(handler func()) {
 // SetupExitSignalHandling sets up a signal channel to listen for termination or interruption signals.
 // When one of these signals is received, all the registered exit handlers will be invoked, just
 // before terminating the program.
-func SetupExitSignalHandling() {
-	setupExitSignalHandling(true)
+func SetupExitSignalHandling(exitChan chan bool) {
+	setupExitSignalHandling(exitChan, true)
 }
 
 // setupExitSignalHandling sets up a signal channel to listen for termination or interruption signals.
 // When one of these signals is received, all the registered exit handlers will be invoked.
 // It is possible to prevent the program from exiting by setting the doExit param to false (used for testing)
-func setupExitSignalHandling(doExit bool) {
+func setupExitSignalHandling(exitChan chan bool, doExit bool) {
```

Comment:
An alternative to this `exitChan` approach would be to pass a `handler func()` similar to what is done with `RegisterExitHandler`. But both approaches are fine, mentioning it here for completeness, changes are not required.

Reply:
I changed the code so it accepts a shutdown function and executes it. This makes testing also easier as I can provide a test implementation of the shutdown func